src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -425,6 +425,28 @@ protected function instantiateObject(array &$data, string $class, array &$contex
                     throw $e;
                 }
 
+                $message = $e->getMessage();
+
+                if (preg_match('/#\d+ \(\$([^)]+)\) must be of type ([^,]+), ([^,]+) given/', $message, $match)) {
+                    $message = sprintf(
+                        'The type of the "%s" parameter for class "%s" must be of type "%s" ("%s" given).',
+                        $match[1],
+                        $class,
+                        $match[2],
+                        $match[3],
+                    );
+                }
+
+                $exception = NotNormalizableValueException::createForUnexpectedDataType(
+                    $message,
+                    $data,
+                    [$class],
+                    $context['deserialization_path'] ?? null,
+                    true,
+                    previous: $e,
+                );
```

Comment:
Perhaps it would be better this way
```suggestion
                $type = $class;
                
                if (preg_match('/#\d+ \(\$([^)]+)\) must be of type ([^,]+), ([^,]+) given/', $message, $match)) {
                    $message = sprintf(
                        'The type of the "%s" parameter for class "%s" must be of type "%s" ("%s" given).',
                        $match[1],
                        $class,
                        $match[2],
                        $match[3],
                    );
                    
                    $type = $match[2];
                }

                $exception = NotNormalizableValueException::createForUnexpectedDataType(
                    $message,
                    $data,
                    [$type],
                    $context['deserialization_path'] ?? null,
                    true,
                    previous: $e,
                );
```